src/Stratis.Bitcoin.Features.Consensus/Rules/CommonRules/PoSTransactionsOrderRule.cs

```diff
@@ -0,0 +1,36 @@
+﻿using Microsoft.Extensions.Logging;
+using NBitcoin;
+using Stratis.Bitcoin.Consensus;
+using Stratis.Bitcoin.Consensus.Rules;
+
+namespace Stratis.Bitcoin.Features.Consensus.Rules.CommonRules
+{
+    /// <summary>Checks that coisntake tx from the proven header is the 2nd transaction in a block and that block has at least 2 transactions.</summary>
+    public class PosTransactionsOrderRule : IntegrityValidationConsensusRule
+    {
+        /// <inheritdoc />
+        /// <exception cref="ConsensusErrors.InvalidTxCount">Thrown in case block has less than 2 transaction.</exception>
+        /// <exception cref="ConsensusErrors.PHCoinstakeMissmatch">Thrown in case coinstake transaction from the proven header missmatches 2nd block transaction.</exception>
+        public override void Run(RuleContext context)
+        {
+            Block block = context.ValidationContext.BlockToValidate;
+
+            if (block.Transactions.Count < 2)
```

Comment:
I think a sync test needs to be done, this will probably fail during the PoW phase of the network. The block may not even have a coinstake.

Reply:
Syncing the networks with the latest fixes